test/unit/math/memory/stack_alloc_test.cpp

```diff
@@ -91,3 +91,13 @@ TEST(stack_alloc,alloc) {
   allocator.recover_all();
 
 }
+
+TEST(stack_alloc, in_stack) {
```

Comment:
I'd suggest doing this in a big loop to make sure we can handle the boundaries across blocks in the memory allocator.  Maybe 100K or something like that would cover everything and still be fast.